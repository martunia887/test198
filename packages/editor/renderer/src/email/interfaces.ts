import { Mark } from 'prosemirror-model';

export type NodeSerializer = (opts: NodeSerializerOpts) => string;
export type MarkSerializer = (opts: MarkSerializerOpts) => string;

export type Style = { [key: string]: string | number | undefined };

export interface NodeSerializerOpts {
  attrs: { [key: string]: any };
  marks: Mark[];
  text?: string | null;
}

export interface MarkSerializerOpts {
  mark: Mark;
  text: string;
}

export interface SmartCardWithUrlAttributes {
  url: string;
}

interface ScData {
  '@type': string;
  generator: {
    '@type': string;
    name: string;
  };
  name: string;
  url: string;
  summary: string;
}

export interface SmartCardWithDataAttributes {
  data: ScData;
}
