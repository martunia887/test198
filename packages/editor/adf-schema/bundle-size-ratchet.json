[
  {
    "id": "src.main",
    "name": "main",
    "stats": {
      "size": 56457,
      "gzipSize": 13737
    }
  },
  {
    "id": "node_modules.main",
    "name": "node_modules [main]",
    "stats": {
      "size": 60668,
      "gzipSize": 18855
    }
  },
  {
    "id": "combined.main",
    "name": "main",
    "stats": {
      "size": 116511,
      "gzipSize": 31932
    }
  }
]