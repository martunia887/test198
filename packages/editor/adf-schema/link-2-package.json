/Users/mdejongh/Projects/Atlassian/atlaskit-mk-2/packages/editor/adf-schema/package.json