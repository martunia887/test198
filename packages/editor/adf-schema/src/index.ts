import vue from 'vue';
import jquery from 'jquery';

export * from './schema';
export * from './utils';

console.log(vue);
console.log(jquery);
