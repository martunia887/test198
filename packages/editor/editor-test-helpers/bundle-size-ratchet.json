[
  {
    "id": "src.main",
    "name": "main",
    "stats": {
      "size": 41802,
      "gzipSize": 12156
    }
  },
  {
    "id": "node_modules.main",
    "name": "node_modules [main]",
    "stats": {
      "size": 1279065,
      "gzipSize": 362042
    }
  },
  {
    "id": "node_modules.async",
    "name": "node_modules [async]",
    "stats": {
      "size": 1402560,
      "gzipSize": 398050
    }
  },
  {
    "id": "atlaskit.core.main",
    "name": "main",
    "stats": {
      "size": 220187,
      "gzipSize": 48596
    }
  },
  {
    "id": "atlaskit.core.async",
    "name": "async",
    "stats": {
      "size": 169481,
      "gzipSize": 32871
    }
  },
  {
    "id": "atlaskit.editor.main",
    "name": "main",
    "stats": {
      "size": 691319,
      "gzipSize": 183385
    }
  },
  {
    "id": "atlaskit.editor.async",
    "name": "async",
    "stats": {
      "size": 38914,
      "gzipSize": 9079
    }
  },
  {
    "id": "atlaskit.elements.main",
    "name": "main",
    "stats": {
      "size": 81579,
      "gzipSize": 20275
    }
  },
  {
    "id": "atlaskit.elements.async",
    "name": "async",
    "stats": {
      "size": 71999,
      "gzipSize": 21515
    }
  },
  {
    "id": "atlaskit.media.main",
    "name": "main",
    "stats": {
      "size": 185456,
      "gzipSize": 43905
    }
  },
  {
    "id": "atlaskit.media.async",
    "name": "async",
    "stats": {
      "size": 923621,
      "gzipSize": 227183
    }
  },
  {
    "id": "atlaskit.teamwork-ecosystem.main",
    "name": "main",
    "stats": {
      "size": 624,
      "gzipSize": 367
    }
  },
  {
    "id": "combined.main",
    "name": "main",
    "stats": {
      "size": 2489996,
      "gzipSize": 680480
    }
  },
  {
    "id": "combined.async",
    "name": "async",
    "stats": {
      "size": 2605270,
      "gzipSize": 691473
    }
  }
]