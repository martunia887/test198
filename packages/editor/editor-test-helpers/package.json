{
  "name": "@atlaskit/editor-test-helpers",
  "version": "9.11.5",
  "description": "Test Helpers for Atlassian Editor components",
  "license": "Apache-2.0",
  "atlaskit:src": "src/index.ts",
  "module": "dist/esm/index.js",
  "main": "dist/cjs/index.js",
  "types": "dist/cjs/index.d.ts",
  "author": "Atlassian Pty Ltd",
  "maintainers": [
    "Stanislas S"
  ],
  "atlaskit": {
    "internal": true,
    "name": "Editor Test Helpers",
    "team": "editor"
  },
  "repository": "https://bitbucket.org/atlassian/atlaskit-mk-2",
  "dependencies": {
    "@atlaskit/adf-schema": "^3.0.0",
    "@atlaskit/analytics-next": "^5.1.2",
    "@atlaskit/editor-common": "^39.17.0",
    "@atlaskit/editor-core": "^112.39.12",
    "@atlaskit/icon": "^19.0.0",
    "@atlaskit/media-core": "^30.0.8",
    "@atlaskit/media-test-helpers": "^24.3.3",
    "@atlaskit/smart-card": "^12.4.1",
    "enzyme": "^3.7.0",
    "prosemirror-commands": "^1.0.7",
    "prosemirror-keymap": "^1.0.1",
    "prosemirror-model": "^1.7.0",
    "prosemirror-state": "^1.2.2",
    "prosemirror-tables": "^0.8.1",
    "prosemirror-utils": "0.9.4",
    "prosemirror-view": "^1.9.12",
    "react-intl": "^2.6.0"
  },
  "peerDependencies": {
    "react": "^16.8.0"
  },
  "devDependencies": {
    "@types/prosemirror-commands": "^1.0.1",
    "@types/prosemirror-keymap": "^1.0.1",
    "@types/prosemirror-model": "^1.5.0",
    "@types/prosemirror-state": "^1.2.0",
    "@types/prosemirror-view": "^1.9.0"
  }
}