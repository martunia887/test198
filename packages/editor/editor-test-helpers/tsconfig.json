{
  "extends": "../../../tsconfig.base.json",
  "compilerOptions": {
    "baseUrl": "./"
  },
  "include": [
    "./src/**/*.ts",
    "./src/**/*.tsx",
    "./tests/**/*.ts",
    "./tests/**/*.tsx",
  ]
}