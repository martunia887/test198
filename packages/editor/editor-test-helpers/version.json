{
	"name": "@atlaskit/editor-test-helpers",
	"version": "8.0.2"
}
