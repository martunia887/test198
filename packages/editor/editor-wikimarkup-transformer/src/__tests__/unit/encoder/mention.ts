import { defaultSchema } from '@atlaskit/adf-schema';
import WikiMarkupTransformer from '../../../index';

import { doc, mention, p } from '@atlaskit/editor-test-helpers';
import { Context } from '../../../interfaces';

describe('ADF => WikiMarkup - Mention', () => {
  const transformer = new WikiMarkupTransformer();

  test('should convert mention node', () => {
    const node = doc(
      p(
        'Hey ',
        mention({ id: 'supertong' })(),
        ', please take a look at this.',
      ),
    )(defaultSchema);
    expect(transformer.encode(node)).toMatchSnapshot();
  });

  test('should convert mention node with context', () => {
    const context: Context = {
      mentionConversion: {
        'abc-123': 'accountid:abc-123',
      },
    };
    const node = doc(
      p('Hey ', mention({ id: 'abc-123' })(), ', please take a look at this.'),
    )(defaultSchema);
    expect(transformer.encode(node, context)).toMatchSnapshot();
  });
});
