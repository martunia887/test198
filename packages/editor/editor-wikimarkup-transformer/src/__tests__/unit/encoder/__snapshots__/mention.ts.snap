// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADF => WikiMarkup - Mention should convert mention node 1`] = `"Hey [~accountid:supertong], please take a look at this."`;

exports[`ADF => WikiMarkup - Mention should convert mention node with context 1`] = `"Hey [~randomPrefix:abc-123], please take a look at this."`;
